mobile/src/main/java/org/openhab/habdroid/util/ExtensionFuncs.kt

```diff
@@ -307,6 +309,15 @@ fun Context.getHumanReadableErrorMessage(url: String, httpCode: Int, error: Thro
     }
 }
 
+fun Context.isDataSaverActive(): Boolean {
+    val dataSaverPref = getPrefs().getBoolean(Constants.PREFERENCE_DATA_SAVER, false)
+    val cm = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    if ((dataSaverPref && cm.isActiveNetworkMetered) || Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
```

Comment:
I think we should take the 'is metered' aspect from the active `Connection`, since we may bind the data traffic to a specific network. We can extract that information from the `NetworkInfo` or `NetworkCapabilities` objects and transfer it to `Connection` from there.
One example for this is 'Wifi without internet' + 'mobile network': active network will report being metered, but we won't transfer data over that interface.
On the other hand, I'm not sure whether we would want data saver if the user uses an unmetered VPN over a metered mobile connection. :man_shrugging:

Reply:
> We can extract that information from the NetworkInfo or NetworkCapabilities objects and transfer it to Connection from there.

:+1: 

> On the other hand, I'm not sure whether we would want data saver if the user uses an unmetered VPN over a metered mobile connection.

The question is if the VPN counts as unmetered if the "carrier" connection is metered.

Reply:
> The question is if the VPN counts as unmetered if the "carrier" connection is metered.

Good point. I just checked, indeed that's the case. VPN over Wifi has the NOT_METERED capability set, VPN over cellular does not.

Reply:
> We can extract that information from the NetworkInfo or NetworkCapabilities objects and transfer it to Connection from there.

I started with this, but I don't know where exactly the information should be added to the connection.

Reply:
@maniac103 Do you have an idea of how to pass the `isMetered` to the connection? If not, I'd remove the whole `isMetered` and

1. activate data saver on every connection if enabled in settings
2. or remove the pref and just respect the device wide data saver.

Reply:
I implemented option 1. The pref turned off is exactly like option 2.